pkgs/build-support/setup-hooks/make-wrapper.sh

```diff
@@ -30,88 +30,96 @@ assertExecutable() {
 makeWrapper() {
     local original="$1"
     local wrapper="$2"
-    local params varName value command separator n fileNames
-    local argv0 flagsBefore flags
+    local -a flagsBefore=()
 
     assertExecutable "$original"
 
     mkdir -p "$(dirname "$wrapper")"
 
-    echo "#! $SHELL -e" > "$wrapper"
+    echo "#! $SHELL -eu" > "$wrapper"
```

Comment:
This is not how people are using the wrapper. They embed arbitrary code with the `--run` directive which often accesses environment variables that may be undefined.

Reply:
Could we `set +u` before the splicing of a `--run`?

Reply:
`--add-flags` may also insert code that refers to undefined variables.

Reply:
Good catch.